dev-lang/lua/lua-5.3.3.recipe

```diff
@@ -0,0 +1,93 @@
+SUMMARY="A powerful, fast, light-weight, embeddable scripting language"
+DESCRIPTION="
+Lua combines simple procedural syntax with powerful data description \
+constructs based on associative arrays and extensible semantics. Lua is \
+dynamically typed, runs by interpreting bytecode for a register-based virtual \
+machine, and has automatic memory management with incremental garbage \
+collection, making it ideal for configuration, scripting, and rapid prototyping.
+"
+HOMEPAGE="http://www.lua.org"
+SOURCE_URI="http://www.lua.org/ftp/lua-$portVersion.tar.gz"
+CHECKSUM_SHA256="5113c06884f7de453ce57702abaac1d618307f33f6789fa870e87a59d772aca2"
+LICENSE="MIT"
+COPYRIGHT="1994-2015, Lua.org, PUC-Rio"
+REVISION="1"
+ARCHITECTURES="x86 x86_gcc2 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	lua$secondaryArchSuffix = $portVersion compat >= 5.3
+	cmd:lua$secondaryArchSuffix = $portVersion compat >= 5.3
+	cmd:lua5.3$secondaryArchSuffix = $portVersion compat >= 5.3
+	cmd:luac$secondaryArchSuffix = $portVersion compat >= 5.3
+	cmd:luac5.3$secondaryArchSuffix = $portVersion compat >= 5.3
+	lib:liblua$secondaryArchSuffix = $portVersion compat >= 5.3
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	cmd:make
+	cmd:gcc$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	lua${secondaryArchSuffix}_devel = $portVersion compat >= 5.3
+	devel:liblua$secondaryArchSuffix = $portVersion compat >= 5.3
+	"
+REQUIRES_devel="
+	lua$secondaryArchSuffix == $portVersion
+	"
+
+PATCHES="lua-$portVersion.patchset"
+if [ $effectiveTargetArchitecture = x86_gcc2 ]; then
+	PATCHES+="
+		lua-$portVersion.gcc2.patchset
+	"
+fi
+
+
+BUILD()
+{
+	make haiku MYCFLAGS="-DLUA_COMPAT_5_1 -DLUA_ROOT='\"$prefix/\"' -DLUA_LDIR='\"$dataDir/lua/5.3/\"'" \
+		LUA_T="lua5.3" LUAC_T="luac5.3" LUA_SO="liblua.so.5.3"
```

Comment:
That's strange. Well, the only possible reason why you could get error (with my previous suggestions) might be that I also dropped the trailing slash at 2 places.
If it does not take too long to build, could you try with one of these 2 alternate suggestions please?
```
	make MYCFLAGS="-DLUA_COMPAT_5_1 -DLUA_ROOT=$prefix/ -DLUA_LDIR=$dataDir/lua/5.3/" \
		haiku LUA_T="lua5.3" LUAC_T="luac5.3" LUA_SO="liblua.so.5.3" \
```
or
```
	make haiku MYCFLAGS="-DLUA_COMPAT_5_1 -DLUA_ROOT=$prefix/ -DLUA_LDIR=$dataDir/lua/5.3/" \
		LUA_T="lua5.3" LUAC_T="luac5.3" LUA_SO="liblua.so.5.3" \
```

Of course, they are equivalent to each other (since what changes is only the order of the parameters), so if one does not work the other won't either.

Hmm, it could be that the contents of these macros are scanned in a strange way, for example to look for the presence of double quotes, but this would be really wrong imho.
So, after all, maybe it's not worth even trying anything ;-) You can probably ignore this whole comment (and the previous too ;-)))